docs/rollout.md

```diff
@@ -0,0 +1,175 @@
+
+Fleet creates up to 50 BundleDeployments at a time,  regardless of the rollout configuration. After creating a batch, Fleet evaluates the deployment status. If the current batch meets readiness criteria, Fleet proceeds to the next batch. Otherwise, it pauses the rollout.
+
+The rollout behavior is configured in the [`rolloutStrategy` in the `fleet.yaml`](ref-fleet-yaml.md)
+
+## How Does Partitioning Work?
+
+Partitions are considered non-ready if any cluster in the partition is not Ready. Deployment continues only if the number of non-ready partitions remains below maxUnavailablePartitions.
+
+:::note
+Fleet deploys up to 50 clusters at a time regardless of partition size.
+:::
+
+![A visual asset displaying flow of rollout in Fleet.](../static/img/flow-rollout-fleet.png)
+
+Various limits that can be configured in Fleet:
+
+| Field | Description | Default |
+| ----- | ----- | ----- |
+| maxUnavailable | Maximum number or percentage of clusters that can be non-ready before halting rollout. | 100% |
+| maxUnavailablePartitions | Number or percentage of partitions that can be non-ready at once. | 0 |
+| autoPartitionSize | Number or percentage of clusters per auto-created partition. | 25% |
+| partitions | Define manual partitions by cluster labels or group. If set, autoPartitionSize is ignored. | – |
+
+Fleet supports both automatic and manual rollout partitioning. For more information, refer to the [rolloutStrategy in  fleet.yaml.](ref-fleet-yaml.md)
+
+**Automatic Partitioning:** Fleet automatically creates partitions using autoPartitionSize.
+
+* If fewer than 200 clusters, Fleet uses a single partition.  
+* If 200+ clusters, partitions are created based on autoPartitionSize.
+
+**Manual Partitioning:** You define specific partitions using the partitions field. This gives you fine-grained control over cluster groupings and rollout order.
+
+:::note
+If you specify partitions manually, the autoPartitionSize is ignored.
+:::
+
+Partitions can be specified by name, `cluster selector`, `cluster group`, and `clusterGroupSelector`.
+
+For example, consider creating a partition named canary for production clusters in a specific group:
+
+```yaml
+rolloutStrategy:
+  partitions:
+    - name: canary
+      maxUnavailable: 10%
+      clusterSelector:
+        matchLabels:
+          env: prod
+      clusterGroup: frontend
+```
+
+Fleet then:
+
+1. Selects clusters based on `clusterSelector`, `clusterGroup`, or `clusterGroupSelector`.  
+2. Starts rollout to the first partition.  
+3. Waits until the partition is **Ready** (or within maxUnavailable threshold).  
+4. Proceeds to the next partition.
+
+### Single Partition Rollout
+
+If you do not define partitions, Fleet creates a single partition with all clusters. Use the `maxUnavailable` setting to limit how many clusters can be non-ready at once. If this limit is exceeded, Fleet pauses the rollout until enough clusters become ready.
```

Comment:
Well, depends on the amount of clusters that are being targeted.

<200: One partition

>=200: Going to be by the default value of `autoPartitionSize` (25%).

Reply:
Just the second sentence, which mentions that. Maybe we just need a bit of rephrasing here.